### PR TITLE
add unit tests for asset package

### PR DESCRIPTION
**Pull request by B-Naoufal**

### CODE REVIEW

File: pkg/assets/asset_test.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	}
+}
+`
+
+func TestLoadAccountsStore(t *testing.T) {
+	// Create a temporary testing JSON file with dummy data
+	tmpFile, err := os.CreateTemp("", "dummy_assets.json")
+	if err != nil {
+		t.Fatalf("Error creating temporary file: %v", err)
+	}
```
Comment (Thykof) [inline pkg/assets/asset_test.go:38]: same for bellow

```suggestion
	assert.NoError(t, err)
```

**Review by Thykof — CHANGES_REQUESTED** [review]
please use `assert` package

File: pkg/assets/asset_test.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	expectedContractAddress := "0x1234567890abcdef"
+	asset, ok := store.Assets[expectedAccountName].ContractAssets[expectedContractAddress]
+	if !ok {
+		t.Fatalf("Expected asset not found in loaded data")
+	}
+
+	// Example: Validate the name of the loaded asset
+	if asset.Name != "DummyToken" {
+		t.Errorf("Unexpected asset name. Got %s, expected %s", asset.Name, "DummyToken")
+	}
```
Comment (Thykof) [inline pkg/assets/asset_test.go:61]:
```suggestion
	assert.Equal(t, asset.Name, "DummyToken")
```

**Review by Thykof — CHANGES_REQUESTED** [review]
please use assert package for all assertion

File: internal/handler/wallet/api_test.go [hunk 3]
```diff
@@ -51,7 +52,12 @@ func MockAPI() (*operations.MassaWalletAPI, prompt.WalletPrompterInterface, *ass
 
 	prompterApp := NewWalletPrompterMock(walletapp.NewWalletApp(), resultChannel)
 
-	AssetsStore, err := assets.NewAssetsStore()
+	assetsJSONPath, err := assets.GetAssetsJSONPath()
+	if err != nil {
+		logger.Warnf("Failed to get AssetsStore JSON file: %v", err)
```
Comment (Thykof) [inline internal/handler/wallet/api_test.go:57]: why log a warning in unit test? should we fail the test if an error occur?

File: pkg/assets/asset_test.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+	store, err := NewAssetsStore(tmpFile.Name())
+	assert.NoError(t, err)
+
+	// Test case 1: Check for an existing asset
+	existingNickname := "dummyAccount"
+	existingContractAddress := "0x1234567890abcdef"
+	exists := store.AssetExists(existingNickname, existingContractAddress)
+	if !exists {
+		t.Errorf("Expected asset to exist, but AssetExists returned false")
+	}
```
Comment (Thykof) [inline pkg/assets/asset_test.go:82]: use assert package

File: pkg/assets/asset_test.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+	// Test case 2: Check for a non-existing asset
+	nonExistingNickname := "nonExistingAccount"
+	nonExistingContractAddress := "0xabcdefabcdefabcdef"
+	notExists := store.AssetExists(nonExistingNickname, nonExistingContractAddress)
+	if notExists {
+		t.Errorf("Expected asset not to exist, but AssetExists returned true")
+	}
```
Comment (Thykof) [inline pkg/assets/asset_test.go:90]: use assert package

File: pkg/assets/asset_test.go [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+	*assetInfo.Decimals = 18
+
+	// Add the asset
+	err = store.AddAsset(nickname, assetAddress, assetInfo)
+	assert.NoError(t, err)
+
+	// Check if the added asset exists
+	if !store.AssetExists(nickname, assetAddress) {
+		t.Errorf("Added asset not found in reloaded store")
+	}
```
Comment (Thykof) [inline pkg/assets/asset_test.go:145]: use assert package

File: pkg/assets/asset_test.go [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+	assert.NoError(t, err)
+
+	// Reload the AssetsStore from the JSON file after deletion
+	deletedStore, err := NewAssetsStore(tmpFile.Name())
+	assert.NoError(t, err)
+
+	// Check if the deleted asset no longer exists
+	if deletedStore.AssetExists(nickname, assetAddress) {
+		t.Errorf("Deleted asset still found in reloaded store")
+	}
```
Comment (Thykof) [inline pkg/assets/asset_test.go:158]: use assert package

**Review by Thykof — CHANGES_REQUESTED** [review]

File: cmd/massa-wallet/massa-wallet.go [hunk 8]
```diff
@@ -27,7 +27,12 @@ func StartServer(app *walletApp.WalletApp) {
 		promptApp = prompt.NewEnvPrompter(app)
 	}
 
-	AssetsStore, err := assets.NewAssetsStore()
+	assetsJSONPath, err := assets.GetAssetsJSONPath()
+	if err != nil {
+		logger.Warnf("Failed to get AssetsStore JSON file: %v", err)
```
Comment (Thykof) [inline cmd/massa-wallet/massa-wallet.go:32]: have you tested this? what happens when assetsJSONPath is not found at line 35 `assets.NewAssetsStore(assetsJSONPath)`

File: pkg/assets/asset_test.go [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+		Decimals: new(int64),
+	}
+	*assetInfo.Decimals = 18
+
+	// Add the asset
+	err = store.AddAsset(nickname, assetAddress, assetInfo)
+	assert.NoError(t, err)
+
+	// Check if the added asset exists
+	assert.True(t, store.AssetExists(nickname, assetAddress), "Added asset not found in reloaded store")
```
Comment (Thykof) [inline pkg/assets/asset_test.go:141]: I don't see where you reload the store

File: pkg/assets/asset_test.go [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+
+	// Test case 2: Delete the added asset and check if it's removed from JSON
+	err = store.DeleteAsset(nickname, assetAddress)
+	assert.NoError(t, err)
+
+	// Reload the AssetsStore from the JSON file after deletion
+	deletedStore, err := NewAssetsStore(tmpFile.Name())
```
Comment (Thykof) [inline pkg/assets/asset_test.go:148]: why do you need to reload the store to take into account the deletion of an asset?

**Review by Thykof — CHANGES_REQUESTED** [review]

File: cmd/massa-wallet/massa-wallet.go (hunk 8 above)
Comment (B-Naoufal) [reply cmd/massa-wallet/massa-wallet.go:32]: GetAssetsJSONPath simply calls AccountPath() which  create the directory if it doesn't exist;

File: pkg/assets/asset_test.go (hunk 9 above)
Comment (B-Naoufal) [reply pkg/assets/asset_test.go:141]: actually we don't need to reload the store , we add an asset and we test if it's added to the store.

Comment (Thykof) [reply pkg/assets/asset_test.go:141]: so if you don't reload the store, please change (or remove) the message "Added asset not found in reloaded store"

File: pkg/assets/asset_test.go [hunk 11]
```diff
@@ -138,12 +138,12 @@ func TestAddAndDeleteAsset(t *testing.T) {
...
-	assert.True(t, store.AssetExists(nickname, assetAddress), "Added asset not found in reloaded store")
+	assert.True(t, store.AssetExists(nickname, assetAddress), "Added asset not found in the store")
 
 	// Test case 2: Delete the added asset and check if it's removed from JSON
 	err = store.DeleteAsset(nickname, assetAddress)
 	assert.NoError(t, err)
 
 	// Check if the deleted asset no longer exists
-	assert.False(t, store.AssetExists(nickname, assetAddress), "Deleted asset still found in reloaded store")
+	assert.False(t, store.AssetExists(nickname, assetAddress), "Deleted asset still found int the store")
```
Comment (Thykof) [inline pkg/assets/asset_test.go:148]:
```suggestion
	assert.False(t, store.AssetExists(nickname, assetAddress), "Deleted asset still found in the store")
```

**Review by Thykof — CHANGES_REQUESTED** [review]

File: pkg/assets/asset.go [hunk 12]
```diff
@@ -37,24 +37,19 @@ type assetData struct {
...
 	}
 
 	return store, nil
 }
 
 // loadaccountsStore loads the data from the assets JSON file into the AssetsStore.
-func (s *AssetsStore) loadaccountsStore() error {
-	assetsJSONPath, err := GetAssetsJSONPath()
-	if err != nil {
-		return errors.Wrap(err, "error getting assets JSON file")
```
Comment (Thykof) [inline pkg/assets/asset.go:55]: you can see here that before, the error was not ignored

**Review by Thykof — CHANGES_REQUESTED** [review]

**Review by Thykof — APPROVED** [review]

**Pull request merged**